File: C11/백준/Silver/11723. 집합/집합.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main(void){
	
	int M;
	scanf("%d", &M);
	
	bool set[21] = {false};
	
	for(int i = 0; i < M; i++){
		char cmd[10];
		int x = 0;
		scanf("%s", cmd);
		
		if (strcmp(cmd, "add") == 0) {
            scanf("%d", &x);
            set[x] = true;
        } else if (strcmp(cmd, "remove") == 0) {
            scanf("%d", &x);
            set[x] = false;
        } else if (strcmp(cmd, "check") == 0) {
            scanf("%d", &x);
            printf("%d\n", set[x] ? 1 : 0);
        } else if (strcmp(cmd, "toggle") == 0) {
            scanf("%d", &x);
            set[x] = !set[x];
        } else if (strcmp(cmd, "all") == 0) {
            for (int j = 1; j <= 20; j++) {
                set[j] = true;
            }
        } else if (strcmp(cmd, "empty") == 0) {
            for (int j = 1; j <= 20; j++) {
                set[j] = false;
            }
        }
	}
	
	return 0;
}

File: Java/백준/Silver/11723. 집합/집합.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int M = Integer.parseInt(br.readLine()); //수행해야 하는 연산의 수
        boolean[] set = new boolean[21];

        for (int i = 0; i < M; i++) {
            String[] input = br.readLine().split(" ");
            String cmd = input[0];

            switch (cmd) {
                case "add" -> {
                    int x = Integer.parseInt(input[1]);
                    set[x] = true;
                }
                case "remove" -> {
                    int x = Integer.parseInt(input[1]);
                    set[x] = false;
                }
                case "check" -> {
                    int x = Integer.parseInt(input[1]);
                    sb.append(set[x] ? "1\n" : "0\n");
                }
                case "toggle" -> {
                    int x = Integer.parseInt(input[1]);
                    set[x] = !set[x];
                }
                case "all" -> {
                    for (int j = 1; j <= 20; j++) {
                        set[j] = true;
                    }
                }
                case "empty" -> {
                    for (int j = 1; j <= 20; j++) {
                        set[j] = false;
                    }
                }
            }
        }

        System.out.print(sb);
    }
}